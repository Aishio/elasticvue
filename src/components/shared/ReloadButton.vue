<template>
  <v-btn flat icon :class="classes" @click="action" :title="title">
    <v-icon>cached</v-icon>
  </v-btn>
</template>

<script>
  export default {
    props: {
      action: {
        type: Function,
        default: () => {
        }
      },
      title: {
        type: String,
        default: 'Reload'
      },
      alignLeft: Boolean
    },
    computed: {
      classes () {
        return {'ml-a': !this.alignLeft}
      }
    }
  }
</script>
