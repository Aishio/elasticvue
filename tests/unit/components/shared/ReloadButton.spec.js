import Vuetify from 'vuetify'
import VBtn from 'vuetify/es5/components/VBtn'
import VIcon from 'vuetify/es5/components/VIcon'
import ReloadButton from '@/components/shared/ReloadButton'
import { createLocalVue, mount } from '@vue/test-utils'

describe('components/shared/ReloadButton.vue', () => {
  let localVue

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(Vuetify, {
      components: {
        VBtn,
        VIcon
      }
    })
  })

  it('should render correct default contents', () => {
    const wrapper = mount(ReloadButton, {
      localVue: localVue
    })

    expect(wrapper.attributes().title).toEqual('Reload')
    expect(wrapper.attributes().id).toEqual('reload-button')
    expect(wrapper.html()).not.toBeNull()
  })

  it('renders a custom title', () => {
    const title = 'myCustom title'
    const wrapper = mount(ReloadButton, {
      localVue: localVue,
      propsData: {
        title
      }
    })

    expect(wrapper.attributes().title).toEqual(title)
  })
})
