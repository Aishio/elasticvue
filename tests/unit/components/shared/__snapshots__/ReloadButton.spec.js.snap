// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/ReloadButton.vue has an empty default action 1`] = `
<div
  class="d-inline-block reload-button"
>
  <button
    class="mr-0 v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--small"
    id="reload-button"
    title="Reload"
    type="button"
  >
    <span
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate mdi mdi-cached theme--light"
      />
    </span>
  </button>
   
  <div
    class="d-inline-block"
    title="Auto reload"
  >
    <div
      class="v-input mt-0 pt-0 d-inline-block v-select--dense-append v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-17"
          class="v-input__slot"
          role="button"
        >
          <div
            class="v-select__slot"
          >
            <div
              class="v-select__selections"
            >
              <input
                aria-readonly="true"
                autocomplete="off"
                id="input-17"
                readonly="readonly"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light "
              style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/shared/ReloadButton.vue should render correct default contents 1`] = `
<div
  class="d-inline-block reload-button"
>
  <button
    class="mr-0 v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--small"
    id="reload-button"
    title="Reload"
    type="button"
  >
    <span
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate mdi mdi-cached theme--light"
      />
    </span>
  </button>
   
  <div
    class="d-inline-block"
    title="Auto reload"
  >
    <div
      class="v-input mt-0 pt-0 d-inline-block v-select--dense-append v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-6"
          class="v-input__slot"
          role="button"
        >
          <div
            class="v-select__slot"
          >
            <div
              class="v-select__selections"
            >
              <input
                aria-readonly="true"
                autocomplete="off"
                id="input-6"
                readonly="readonly"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light "
              style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
